### DISCUSSION TITLE
Install instructions - Start the app

### DISCUSSION THREAD

**Discussion by markusthiel** [photobooth-app/photobooth-app#189] [General]
Install instructions semm to be not correct:

```
**First start**
Create data folder
The photobooth-app automatically uses the current folder as data folder. All images, logs and config files will be stored in this folder.

mkdir ~/photobooth-data
Start the app

cd ~/photobooth-data
Following only if installed via venv method

source ~/photobooth-app/myenv/bin/activate
Start app. Current dir will be used as working-directory!

photobooth
```

After creating a folder and changing into it, it is of course empty. What am i missing here?

**Comment by mgineer85** [comment]
the last line is the comnand to Start the app. on start it creates the Files and Folders in the current dir.
Maybe it's as simple as that?

**Comment by markusthiel** [comment]
`xxx@photobooth:~/.local/pipx/venvs/photobooth-app $ mkdir ~/photobooth-data

xxx@photobooth:~/.local/pipx/venvs/photobooth-app $ cd ~/photobooth-data

xxx@photobooth:~/photobooth-data $ photobooth

-bash: photobooth: command not found

xxx@photobooth:~/photobooth-data $`

**Reply by markusthiel** [reply]
Ok i missed something: i had to open a new ssh window, thanks!

**Reply by mgineer85** [reply]
Which platform you use?
Did you restart after installation? Could be required if it’s the first pipx installation

**Comment by markusthiel** [comment]
i installed it on an raspberry pi 5. It was indeed the first pipx install.

Is there a way to automatically display the last taken photo in fullscreen until the next photo is taken? For example, on a projector, so that at an event the newest photo is always visible to everyone?

**Reply by mgineer85** [reply]
Until today this is not available.
In v1 there is the standalone gallery available ( 1 item in the list here https://photobooth-app.org/reference/foldersandurls/)
In v2 this can be extended. Please file a feature request if you like or even implement and send a pull request if you would like to contribute 😌

**Reply by mgineer85** [reply]
Ah, there is also the rest api available which you could use to build your own solution

**Reply by markusthiel** [reply]
Thanks!

**Comment by mathiasonia** [comment]
Hi I'm french future user of the photobooth-app and I'm newbie with raspberry and python language 

I try to install first on a rasperry pi3B+ and at the end I see it wasn't really good on a pi3, I will install on a pi4 next day 

for the Pi3 I make step by step installation and I do not have anything when I start to launch application 
at this step

"Start app. Current dir will be used as working-directory!

photobooth"

have you some tips to check if all soft or library are install
or a really newbie step-by-step installation

thanks for all

**Reply by Striker62** [reply]
for me to start the application only the "photobooth" did not work.
I start the app with "python3 -m photobooth" that worked for me

do not forget to cd in the pgotobooth-app dir

**Reply by mgineer85** [reply]
It could be that "photobooth" command needs a restart, but usually it should work.
@mathiasonia  I saw you went into proofreading the translation. Would be very welcome, thank you very much!

**Reply by mathiasonia** [reply]
I try to make an installation on a new format SDcard but I have a message at the end with 

maxence@rpimathias:~/photobooth-data $ python3 -m photobooth
/usr/bin/python3: No module named photobooth

**Reply by mgineer85** [reply]
Did you try just „photobooth“ after a restart? 
which system exactly and which installation method?

**Reply by mgineer85** [reply]
You ran `pipx ensurepath `?

**Reply by mathiasonia** [reply]
I'm with a rpi 3 B+

to install I follow the page "installation" 
I start with this part 

"System Preparation

RaspberryPi OS
On a fresh Raspberry Pi OS 64bit, run following commands:

Update System (RPi)

sudo apt update
sudo apt upgrade
Tweak system settings (RPi)
To use hardware input from keyboard or presenter, the current user needs to be added to tty and input group.


sudo usermod --append --groups tty,input $(whoami)
You also might want to [check some display settings](https://photobooth-app.org/extras/display/).

Install system dependencies (RPi)
Following dependencies to be installed for Raspberry Pi OS 64bit. Adjust for debian/ubuntu. Picamera2 is only available on Raspberry Pi.


sudo apt -y install ffmpeg libturbojpeg0 python3-pip libgl1 libgphoto2-dev fonts-noto-color-emoji rclone inotify-tools "

and after method C

but in the middle I have an error with 
"building wheel for opencv-python (pyproject.toml)

**Reply by mgineer85** [reply]
In latest bookworm you are not supposed to use method c. It should warn you about that.
Please use method A.
And double check that you really use 64Bit and bookworm (no legacy)

**Reply by mathiasonia** [reply]
I bought a rpi 4 to install bookworm 64 bit and test as soon as I found a micro HDMI adapter

for the moment I try with a rpi 3 to learn how this machine work I 'm not programmer just curious and I used 15 year ago a handmade photobooth with a camera and a car screen with remote control but now it works bad.

I just want a solution like yours you "plug and shoot"

I want to use it with a computer screen 24" 


last question I can try the A method directly or I need to uninstall something

**Comment by Striker62** [comment]
I Double Check how I start my app.
python -m photobooth 

Not Python3

**Reply by mathiasonia** [reply]
Hi when I wrote like you I have 
"(myenv) maxence@rpimathias:~/photobooth-data $ python -m photobooth

**/home/maxence/myenv/bin/python: No module named photobooth**
(myenv) maxence@rpimathias:~/photobooth-data $ "

**Reply by Striker62** [reply]
You're ussing the python virtual env. sorry I do not have any experience with this kind of installtion .
But I'll do some reseach and come back if I find some thing for you

**Reply by mathiasonia** [reply]
I not really sure to make the good thing but I have the same result if I follow the method B 

I'm not sure to make the good thing to have the application works and maybe when I paste the command line I forgot some word

but I want to learn how to make this application works, I 'm just a end-user but I'm curious

**Reply by Striker62** [reply]
at first I tried to run the app on a raspberry pi 4 bookworm. but I could not make it run.
so I decide to use Bullseye on my pi 4 and did not install in venv.
I know that it is a bit tricky to install python stuff on bookworm without venv.
I'm  more familiar to run python without venv,

**Comment by mathiasonia** [comment]
a question - can I start installation and try by ssh I received my micro hdmi cable tomorrow and I want to try this photobooth application

thank you for your advice, I'm a really beginner, I will try this evening, if it's possible to install on a rpi 4 with bulleyes 64bit and see what's happen

**Comment by Striker62** [comment]
I setup all my raspberry ' s with the "raspberry pi imager" there I have a default stored to have e.g. wlan enabled with ssid ,and also enable ssh.
I do the whole installation normaly via ssh . If you have the installation successfull managed you can use the app via the browser of your pc.
http://<hostname>:8000 .

I attched a monitor to my Raspbbery after I got everything working.

**Reply by Striker62** [reply]
http://"hostname":8000 .

**Comment by mathiasonia** [comment]
Good evening 

I finally make a new installation by ssh and now all works fine with a camera module 3 on a RPI 4 64bit with bulleye

thanks a lot for your patience

I will make some test in a few day

**Reply by mgineer85** [reply]
Please note that you use a Legacy system. The picamera software does not get any updates on the old system any more. You will run into bugs and they are not fixed.
I highly recommend current versions.

**Comment by Striker62** [comment]
I've setup a new photobooth with Bookworm but not in venv .
If you also want to install global you only have to add "--break-system-packages"  after the command form method C.

Will do more tests , if everthing works fine I'll stick to bookworm.

**Reply by mathiasonia** [reply]
thanks I will make a test when I found time

**Comment by mathiasonia** [comment]
always with installation I try to make an autostart

but I have no result because I can't copy or create the file into the autostart folder

 "Error saving file.
 Error opening file"/etc/xdg/autostart:photobooth-app.desktop":
 
 The file on disk may now truncated!"

**Reply by Striker62** [reply]
did you try with sudo ?

sudo nano etc/xdg/autostart/photobooth-app.desktop
